numpy/_core/meson.build

```diff
@@ -128,6 +128,16 @@ if use_intel_sort and not fs.exists('src/npysort/x86-simd-sort/README.md')
   error('Missing the `x86-simd-sort` git submodule! Run `git submodule update --init` to fix this.')
 endif
 
+# Setup openmp flags for x86-simd-sort:
+omp_cflags = []
+omp = []
+if use_intel_sort and cpp.has_argument('-fopenmp')
+  omp = dependency('openmp', required : false)
+  if omp.found()
+    omp_cflags = ['-fopenmp', '-DXSS_USE_OPENMP']
+  endif
+endif
```

Comment:
Are we "all good" to use OpenMP in NumPy directly? I thought there were other ecosystem interaction concerns to consider, like cross-interactions with OpenBLAS or wheel-related issues, etc. Maybe this is just for a custom local build rather than for activation in wheels.

If we are actually "ok" for that, I guess that in addition to the env variable Sayed mentioned there is also `threadpoolctl` where one might need to modulate both with something like `with controller.limit(limits={"openblas": 2, "openmp": 4})`?

Reply:
> Are we "all good" to use OpenMP in NumPy directly?

I am not familiar with implications of using openMP and how it could potentially interact with other modules. I was hoping to get an answer to that via the pull request and everyone's input.

Reply:
OpenBLAS usually manages OpenMP itself, which creates a bit of confusion when nesting it.

OpenBLAS now has:
https://github.com/OpenMathLib/OpenBLAS/blob/develop/driver/others/blas_server_callback.c

Which we can use if we have a central thread pool we want to re-use across multiple things?

Reply:
@ogrisel I was wondering if you know anything about this (i.e. whether it is safe to use OpenMP in NumPy, or it would create issues).

Reply:
> Which we can use if we have a central thread pool we want to re-use across multiple things?

The `libopenblas` we ship inside our wheels is always built with pthreads, not openmp. Build scripts live at https://github.com/MacPython/openblas-libs/tree/main/tools.

Reply:
scikit-learn does run into issues with running OpenMP's threadpool together with OpenBLAS: https://github.com/scikit-learn/scikit-learn/issues/28883

There is a draft PR here to force OpenBLAS to use OpenMP (alway from pthreads): https://github.com/scikit-learn/scikit-learn/pull/29403

Reply:
> I thought there were other ecosystem interaction concerns to consider, like cross-interactions with OpenBLAS or wheel-related issues 

Yes, there are some quite ugly issues with multiple installed openmp libraries and segfaults depending on import order, see
https://github.com/microsoft/LightGBM/issues/6595